ClientLib/src/main/java/org/droidplanner/services/android/impl/core/drone/variables/HeartBeat.java

```diff
@@ -64,8 +64,8 @@ public short getMavlinkVersion() {
     public void onHeartbeat(MAVLinkMessage msg) {
         msg_heartbeat heartBeatMsg = msg instanceof msg_heartbeat ? (msg_heartbeat) msg : null;
         if(heartBeatMsg != null){
-            sysid = (byte) msg.sysid;
-            compid = (byte) msg.compid;
+            sysid  = validateToUnsignedByteRange(msg.sysid);
```

Comment:
I added this check, so that an invalid conversion would break, but since this comes over the wire it would be safe to cast from int to short. I think where the java library goes wrong is that it uses int in the first place. if it used short these issue would have shown up, as there would be no need to cast and therefore mask the problem.